pkg/util/ciutil/generic.go

```diff
@@ -0,0 +1,36 @@
+// Copyright 2016-2019, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package ciutil
+
+import (
+	"os"
+)
+
+// genericCICI represents a generic CI/CD system
+// that belongs to the Azure DevOps product suite.
+type genericCICI struct {
```

Comment:
I assume just `genericCI` would be sufficient? (Same with `circleCICI`.)

Also, the comment mentions Azure DevOps...

Reply:
There is also a constant called `GenericCI` which represents the name of the CI system. Same with Circle CI. These two are unfortunately have the acronym CI in their name, which makes it a bit tricky.